src/ray/core_worker/core_worker.cc

```diff
@@ -896,6 +898,16 @@ Status CoreWorker::SubmitActorTask(const ActorID &actor_id, const RayFunction &f
   return status;
 }
 
+Status CoreWorker::KillTask(const ObjectID &object_id) {
+  auto task_id = object_id.TaskId();
+  if (task_manager_->IsTaskPending(task_id)) {
+    auto task_spec = task_manager_->GetTaskSpec(object_id.TaskId());
+    if (!task_spec.IsActorCreationTask())
```

Comment:
let's do a RAY_CHECK instead of if statement here because this is definitely checked by python code.

Reply:
I don't think it actually is.